grails-app/conf/BuildConfig.groovy

```diff
@@ -125,11 +125,16 @@ grails.project.dependency.resolution = {
          */
         test("com.icegreen:greenmail:1.5.10") { excludes "junit" }
 
-        // Required for GPars
+        build 'org.codehaus.gpars:gpars:0.12'  // otherwise early build chain uses 0.9
```

Comment:
Am I misunderstanding the scopes? Does scope work upwards or downwards? i.e. compile -> runtime or vice versa?

Reply:
In other words, do we need boht build and compile? Or just build?

Reply:
My understanding is that everything in the `compile` class path is also included in the `runtime` class path, but the `build` class path is disjoint from both.

In other words:
- I need the dependency at runtime only -> `runtime`
- I need the dependency at compile time and runtime -> `compile`
- I only need the dependency at compile time -> `implementation` (Gradle 4+)
- I need the dependency at compile time and runtime and Grails also silently uses it -> `build` _and_ `compile`.

If you run `grails dependencyReport` before and after this PR you can see that the build step will use GPars 0.9 while the others use 0.12. That seemed like a recipe for trouble, but I can't point to a specific problem it was causing.